src/Tribe/Editor/Provider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+class Tribe__Events__Editor__Provider extends tad_DI52_ServiceProvider {
+
+	/**
+	 * Binds and sets up implementations.
+	 *
+	 * @since  TBD
+	 *
+	 */
+	public function register() {
+		$this->container->singleton( 'tec.editor', 'Tribe__Events__Editor' );
+
+		// Should we continue loading?
+		if ( ! tribe( 'tec.editor' )->is_gutenberg_active() ) {
```

Comment:
should we also limit our scope to being on a TEC editor screen?

Reply:
Yes and No. Parts of it need to load on the front-end for rendering blocks.

Reply:
Ya this is the interesting part of this game, especially as we get into complex blocks. Organizing code, especially the pcss/js that drives a block on fe and in edit mode in an efficient way is not something i have fully sorted yet on the agency side. working that through this week.